x-pack/plugins/security_solution/public/common/components/sourcerer/readme.md

```diff
@@ -0,0 +1,53 @@
+# Sourcerer Component
+
+
+### Adding sourcerer to a new page
+- In order for the sourcerer to show up on a page, it needs to be added to the array `sourcererPaths` in `containers/sourcerer/index.tsx`
+- The scope of a sourcerer component will be default unless the path is added to the `detectionsPaths` array, in which case the scope can be detections in `containers/sourcerer/index.tsx`
+
+## Default Sourcerer
+![](../../images/default.png)
+- The data view is shown in the dropdown. All index aliases that have existing indices are selected by default. Index aliases that do NOT have existing indices are shown in the dropdown as deactivated
+
+## Detections Sourcerer
+![](../../images/alerts.png)
+- The detections sourcerer is locked to the signals index, and has an "Alerts" flag
+
+## Timeline Sourcerer
+![](../../images/timeline.png)
+- The timeline sourcerer is almost exactly like the default sourcerer, but it has a checkbox to "Show only detection alerts" which will automatically select the default data view with the alerts index locked in:
+![](../../images/timeline-alerts.png)
+
+## Sourcerer Quirks
+ - Signals index is not always defined yet, so we `pollForSignalIndex` on mount of detections sourcerer and timeline sourcerer
+ - Signals index is hidden in default scope as it is never used on our Explore pages where default scope is used
+ - When `selectedPatterns` changes, we updated the `onUpdateDetectionAlertsChecked` boolean
+ - Reset button sets to the active `patternList` on the default security data view
+ - The "Modified" label gets applied when the data view `selectedPatterns` diverge from the `patternList`. It does not appear when changing data views and using that data views patternList in full. It does not appear until Save is pressed!
+   ![](../../images/timeline_alerts.png)
+
+## Legacy Timeline Sourcerer
+ - "Update available" workflow for pre-8.0 timelines and timelines that have indices that got deleted from the default data view
+
+### Flow 1
+**Note:** i paused the most in this flow so y'all can ready the copy, the rest of the flows go quicker
+Legacy Timeline includes an active index pattern that is not included in the default data view, user decides to update data view. Page refresh is prompted
```

Comment:
Does this create a new data view that now includes the active index that was not included in the data view? Or by "not included" are you saying the index is part of the data view, but just was hidden or something by default?

Reply:
yes it does include that, that is what Flow 6 describes. not included meaning the index pattern is not in the data view